### PR TITLE
extract client for azure strg impl

### LINKED ISSUE #292: Extract client iface for Azure backed Storage
To enable unit tests and for consistency. ref: #254

### PR DESCRIPTION
**Pull request by marpio**
Fixes #292
To make the cdn/blob storage implementations consistent (ref: #254 and #280) and enable unit tests. Also goes hand in hand with #264

### CODE REVIEW

File: pkg/storage/azurecdn/storage.go [hunk 1]
```diff
@@ -83,58 +75,16 @@ func (s *Storage) Save(ctx context.Context, module, version string, mod []byte,
...
-		_, err := azblob.UploadStreamToBlockBlob(ctx, stream, url, uploadStreamOpts)
-		return err
-	}
-
+func (s *Storage) uploadStream(ctx context.Context, errChan chan<- error, module, version, ext string, stream io.Reader, contentType string) {
 	select {
-	case errChan <- save():
+	case errChan <- s.cl.UploadWithContext(ctx, config.PackageVersionedName(module, version, ext), contentType, stream):
 	case <-ctx.Done():
 		errChan <- fmt.Errorf("uploading %s/%s.%s timed out", module, version, ext)
```
[the remark this answers is not in the document]
Comment (marpio) [reply pkg/storage/azurecdn/storage.go]: is fixed in #264

File: pkg/storage/azurecdn/storage.go [hunk 2]
```diff
@@ -37,43 +52,20 @@ func New(accountName, accountKey string) (*Storage, error) {
...
-	}
-	uploadOpts := func(contentType string) uploadOptions {
-		return uploadOptions{
-			accessConditions: emptyBlobAccessCond,
-			metadata:         emptyMeta,
-			blobHTTPHeaders:  httpHeaders(contentType),
-		}
-	}
-
 	const numUpload = 3
```
[the remark this answers is not in the document]
Comment (marpio) [reply pkg/storage/azurecdn/storage.go]: will decide in #264 since this will go away from save

**Comment by robjloranger** [comment]
LGTM

**Comment by robjloranger** [comment]
If you're happy @marwan-at-work this can go in :)

**Comment by marwan-at-work** [comment]
@marpio LGTM, just update the branch and merge :) -- @robjloranger do we usually update branch from the github ui or do we rebase from the git cli?

**Comment by robjloranger** [comment]
Either or, I usually ask the person to do it because it's always not fun when your the one working on something and you are unaware that the branch has changed on your fork.

In this case I just did it to save @marpio some time

**Comment by arschles** [comment]
fwiw, I do the same via the GH API so folks don't have to come back and update their branch before they can get their code reviewed again

**Pull request merged**